packages/drivers/routerlicious-driver/src/deltaStorageService.ts

```diff
@@ -18,37 +18,35 @@ export class DocumentDeltaStorageService implements IDocumentDeltaStorageService
     constructor(
         private readonly tenantId: string,
         private readonly id: string,
-        private readonly tokenProvider: api.ITokenProvider,
         private readonly storageService: IDeltaStorageService) {
     }
 
     public async get(from?: number, to?: number): Promise<api.ISequencedDocumentMessage[]> {
-        return this.storageService.get(this.tenantId, this.id, this.tokenProvider, from, to);
+        return this.storageService.get(this.tenantId, this.id, from, to);
     }
 }
 
 /**
  * Provides access to the underlying delta storage on the server for routerlicious driver.
  */
 export class DeltaStorageService implements IDeltaStorageService {
-    constructor(private readonly url: string) {
+    constructor(private readonly url: string, private readonly tokenProvider: ITokenProvider) {
     }
 
     public async get(
         tenantId: string,
         id: string,
-        tokenProvider: api.ITokenProvider,
         from?: number,
         to?: number): Promise<api.ISequencedDocumentMessage[]> {
         const query = querystring.stringify({ from, to });
 
         let headers: { Authorization: string } | null = null;
 
-        const token = (tokenProvider as TokenProvider).token;
+        const storageToken = await this.tokenProvider.fetchStorageToken();
```

Comment:
I think you need to start wrapping these with telemetry (at some level) - it's the first question that people ask when looking at perf - what is time break down between token fetch and actual call?

Reply:
Yes I have seen that in ODSP driver. Opened an issue to track that (#3959)